### PR TITLE
Add provenance for PemReader

### LINKED ISSUE #28613: [FEATURE REQ] Add some doc comments to PemReader to explain some background as to why it exists
### Library name

Azure.Core

### Please describe the feature.

n/a

### PR DESCRIPTION
**Pull request by heaths**
Fixes #28613

### CODE REVIEW

File: sdk/core/Azure.Core/src/Shared/PemReader.cs [hunk 1]
```diff
@@ -12,6 +12,12 @@ namespace Azure.Core
     /// <summary>
     /// Reads PEM streams to parse PEM fields or load certificates.
     /// </summary>
+    /// <remarks>
+    /// This class provides a downlevel PEM decoder since <c>PemEncoding</c> wasn't added until net5.0.
+    /// The <c>PemEncoding</c> class takes advantage of other implementation changes in net5.0 and,
+    /// based on conversations with the .NET team, runtime changes. This was based on <c>PemEncodiner</c>
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/src/Shared/PemReader.cs:18]:
```suggestion
    /// based on conversations with the .NET team, runtime changes. This was based on <c>PemEncoding</c>
```

File: sdk/core/Azure.Core/src/Shared/PemReader.cs [hunk 2]
```diff
@@ -12,6 +12,12 @@ namespace Azure.Core
...
+    /// and reviewed by @bartonjs on the .NET / cryptography team.
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/src/Shared/PemReader.cs:19]: Should we be putting GitHub aliases in ref docs?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/core/Azure.Core/src/Shared/PemReader.cs (hunk 2 above)
Comment (heaths) [reply sdk/core/Azure.Core/src/Shared/PemReader.cs:19]: Perhaps not in rendered doc comments. As just a code comment sure. I think it's important to bring up here as bartonjs is the .NET security expert worth mentioning in this regard. Maybe I could refactor that a bit.

File: sdk/core/Azure.Core/src/Shared/PemReader.cs (hunk 1 above)
Comment (heaths) [reply sdk/core/Azure.Core/src/Shared/PemReader.cs:18]: Ugh. I kept wanting to type PemEncoder even though I knew it was PemEncoding. Guess I got my wires crossed here.

**Pull request merged**